staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/testing/testcase.go

```diff
@@ -1051,3 +1051,89 @@ func NewObjectInterfacesForTest() admission.ObjectInterfaces {
 	corev1.AddToScheme(scheme)
 	return admission.NewObjectInterfacesFromScheme(scheme)
 }
+
+// PrependIsReadyValidatingWebhook prepends a is-ready webhook to the input hooks, which can be used
+// as a marker to verify webhook registeration is finished.
+func PrependIsReadyValidatingWebhook(url *url.URL, hooks []registrationv1.ValidatingWebhook) []registrationv1.ValidatingWebhook {
+	policyFail := registrationv1.Fail
+	ccfgURL := urlConfigGenerator{url}.ccfgURL
+
+	isReadyHook := registrationv1.ValidatingWebhook{
+		Name: "is-webhook-configuration-ready.k8s.io",
+		// Rules: matchEverythingRules,
+		Rules: []registrationv1.RuleWithOperations{{
+			Operations: []registrationv1.OperationType{registrationv1.OperationAll},
+			Rule: registrationv1.Rule{
+				APIGroups:   []string{""},
+				APIVersions: []string{"v1"},
+				Resources:   []string{"configmaps"},
+			},
+		}},
+		NamespaceSelector:       &metav1.LabelSelector{},
+		ObjectSelector:          &metav1.LabelSelector{},
+		FailurePolicy:           &policyFail,
+		AdmissionReviewVersions: []string{"v1beta1"},
+	}
+	if hooks[0].ClientConfig.Service != nil {
+		isReadyHook.ClientConfig = ccfgSVC("disallowReason")
+	} else {
+		isReadyHook.ClientConfig = ccfgURL("disallowReason")
+	}
+
+	ret := []registrationv1.ValidatingWebhook{isReadyHook}
+	for _, hook := range hooks {
+		ret = append(ret, hook)
+	}
+	return ret
+}
+
+// PrependIsReadyMutatingWebhook prepends a is-ready webhook to the input hooks, which can be used
+// as a marker to verify webhook registeration is finished.
+func PrependIsReadyMutatingWebhook(url *url.URL, hooks []registrationv1.MutatingWebhook) []registrationv1.MutatingWebhook {
+	policyFail := registrationv1.Fail
+	ccfgURL := urlConfigGenerator{url}.ccfgURL
+
+	isReadyHook := registrationv1.MutatingWebhook{
+		Name: "is-webhook-configuration-ready.k8s.io",
+		// Rules: matchEverythingRules,
+		Rules: []registrationv1.RuleWithOperations{{
+			Operations: []registrationv1.OperationType{registrationv1.OperationAll},
+			Rule: registrationv1.Rule{
+				APIGroups:   []string{""},
+				APIVersions: []string{"v1"},
+				Resources:   []string{"configmaps"},
+			},
+		}},
+		NamespaceSelector:       &metav1.LabelSelector{},
+		ObjectSelector:          &metav1.LabelSelector{},
```

Comment:
I expected a namespace or object selector here to scope the interception so that it wouldn't disrupt any other requests (and to label the object in NewMarkerAttribute similarly).  applies to PrependIsReadyValidatingWebhook as well

Reply:
Done. Added a label "webhook-ready-marker": "true"